internal/buffer/buffer.go

```diff
@@ -690,6 +690,7 @@ func parseDefFromFile(f config.RuntimeFile, header *highlight.Header) *highlight
 	}
 
 	if header == nil {
+		// TODO: use faster highlight.MakeHeader() when possible
```

Comment:
But `highlight.MakeHeader()` can only be used, when we iterate over `config.RTSyntaxHeader`, otherwise we can't guarantee, that we access the correct lines or am I wrong?
So in case of `findRuntimeSyntaxDef()` we could iterate over the `config.RTSyntaxHeader` first, to create the `header` structure and then iterate over the normal `config.RTSyntax` and passing the already created `header` to `parseDefFromFile()`.

Reply:
Yes, so my point was that we should try to use `config.RTSyntaxHeader` whenever possible, to make it faster. That's what it is for.

Actually later I've realized that this should not have much impact on performance (for now): in `findRuntimeSyntaxDef("default", nil)` we're gonna call this `MakeHeaderYaml()` just once (for the file that satisfies the condition `f.Name() == "default"`), not hundreds of times.

So I think I'll remove this TODO. All this still looks messy, but I guess we should rather start cleaning this mess by figuring out how to fix the buggy include logic (per below comments) without making the code unreadable again...